src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/EligibilityServiceTest.kt

```diff
@@ -14,232 +16,372 @@ import java.time.ZoneId
-        recall = null,
-      ),
-    )
-    assertThat(result).containsExactly("has no conditional release date")
-  }
+    @Test
+    fun `Recall flag is null in NOMIS - not ineligibile due to recall`() {
+      val result = service.getIneligibilityReasons(
+        aPrisonerSearchResult.copy(
+          conditionalReleaseDate = null,
+          recall = null,
+        ),
+      )
+      assertThat(result).containsExactly("has no conditional release date")
+    }
 
-  @Test
-  fun `Person has no ARD and no CRD - not eligible for CVL `() {
-    val result = service.getIneligibilityReasons(
-      aPrisonerSearchResult.copy(
-        confirmedReleaseDate = null,
-        conditionalReleaseDate = null,
-      ),
-    )
-    assertThat(result).containsExactly("has no conditional release date")
-  }
+    @Test
+    fun `Person has no ARD and no CRD - not eligible for CVL `() {
+      val result = service.getIneligibilityReasons(
+        aPrisonerSearchResult.copy(
+          confirmedReleaseDate = null,
+          conditionalReleaseDate = null,
+        ),
+      )
+      assertThat(result).containsExactly("has no conditional release date")
+    }
 
-  @Test
-  fun `Person with imprisonmentStatus ACTIVE IN - eligible for CVL `() {
-    val result = service.getIneligibilityReasons(
-      aPrisonerSearchResult.copy(
-        imprisonmentStatus = "ACTIVE IN",
-      ),
-    )
-    assertThat(result).isEmpty()
+    @Test
+    fun `Person with imprisonmentStatus ACTIVE IN - eligible for CVL `() {
+      val result = service.getIneligibilityReasons(
+        aPrisonerSearchResult.copy(
+          imprisonmentStatus = "ACTIVE IN",
+        ),
+      )
+      assertThat(result).isEmpty()
+    }
+
+    @Test
+    fun `Person with imprisonmentStatus BOTUS - not eligible for CVL `() {
+      val result = service.getIneligibilityReasons(
+        aPrisonerSearchResult.copy(
+          imprisonmentStatus = "BOTUS",
+        ),
+      )
+      assertThat(result).containsExactly("is breach of top up supervision case")
+    }
   }
 
-  @Test
-  fun `Person with imprisonmentStatus BOTUS - not eligible for CVL `() {
-    val result = service.getIneligibilityReasons(
-      aPrisonerSearchResult.copy(
-        imprisonmentStatus = "BOTUS",
-      ),
-    )
-    assertThat(result).containsExactly("is breach of top up supervision case")
+  @Nested
+  inner class PrrdCases {
+    @BeforeEach
+    fun setup() {
+      service = EligibilityService(clock, true)
+    }
+
+    @Test
+    fun `Person is eligible for CVL`() {
+      val result = service.isEligibleForCvl(aRecallPrisonerSearchResult)
+      assertThat(result).isTrue()
+    }
+
+    @Test
+    fun `Person is ineligible if PRRD is null`() {
+      val result = service.getIneligibilityReasons(
+        aRecallPrisonerSearchResult.copy(
+          postRecallReleaseDate = null,
+        ),
+      )
+      assertThat(result).containsExactly("has no conditional release date", "has no post recall release date")
+    }
```

Comment:
assertThat(result).contains("has no post recall release date")
your not really testing crd are you?

Reply:
if we're evaluating PRRD then it means some CRD rule will have failed so it will always have at least one CRD failure along with the PRRD failure

Reply:
Are I knew I was missing something!